### PR TITLE
Add filter iterators for weak and blueprints.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review
@arnej27959 : FYI

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍

**Review by geirst — APPROVED** [review]
👍

File: searchlib/src/vespa/searchlib/attribute/attribute_blueprint_factory.cpp [hunk 1]
```diff
@@ -416,9 +416,21 @@ class DirectWandBlueprint : public queryeval::ComplexLeafBlueprint
...
 
+std::unique_ptr<SearchIterator>
+DirectWandBlueprint::createFilterSearch(bool, FilterConstraint) const
+{
+    std::vector<DocumentWeightIterator> iterators;
+    iterators.reserve(_terms.size());
+    for (const IDocumentWeightAttribute::LookupResult &r : _terms) {
+        _attr.create(r.posting_idx, iterators);
+    }
+    return attribute::DocumentWeightOrFilterSearch::create(std::move(iterators));
```
Comment (arnej27959) [inline searchlib/src/vespa/searchlib/attribute/attribute_blueprint_factory.cpp:431]: when constraint is LOWER_BOUND, all WeakAnd variants must create filter as if it's AND.

Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/attribute_blueprint_factory.cpp:431]: I'll change it to use EmptySearch for LOWER_BOUND.

Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/attribute_blueprint_factory.cpp]: Fixed.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**